### PR TITLE
Add Neovim Nightly version 0.8.0-dev-1092-g89b9eab63

### LINKED ISSUE #79698: [Package Request]: Add Neovim Nightly
### Package Requested

- [ ] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [X] I have checked that the installer for this package is not contained in a .zip file
- [ ] I think there is a new version available and I have provided the URL.

### Please provide the following information

Publisher:  neovim.io
Publisher Url:  https://neovim.io
Package Name:  Neovim Nightly
Description:  Vim-fork focused on extensibility and usability
Package Version:  0.8.0-dev-1092-g89b9eab63
Installer URL:  https://github.com/neovim/neovim/releases/download/nightly/nvim-win64.msi

Current Package Identifier (for new versions):

### PR DESCRIPTION
**Pull request by trbrink**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

Closes #79698 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/79699)

### CODE REVIEW

File: manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.2.0.schema.json
+
+PackageIdentifier: Neovim.Neovim.Nightly
+PackageVersion: 0.8.0-dev-1092-g89b9eab63
```
Comment (vedantmgoyal2009) [inline manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.installer.yaml:5]:
```suggestion
PackageVersion: 0.8.0
```

File: manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.2.0.schema.json
+
+PackageIdentifier: Neovim.Neovim.Nightly
+PackageVersion: 0.8.0-dev-1092-g89b9eab63
```
Comment (vedantmgoyal2009) [inline manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.yaml:5]:
```suggestion
PackageVersion: 0.8.0
```

File: manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.locale.en-US.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Created using wingetcreate 1.1.2.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.2.0.schema.json
+
+PackageIdentifier: Neovim.Neovim.Nightly
+PackageVersion: 0.8.0-dev-1092-g89b9eab63
```
Comment (vedantmgoyal2009) [inline manifests/n/Neovim/Neovim/Nightly/0.8.0-dev-1092-g89b9eab63/Neovim.Neovim.Nightly.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 0.8.0
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]
Rename version folder also.

**Comment by trbrink** [comment]
@vedantmgoyal2009, I made the changes you request.

**Review by KaranKad — APPROVED** [review]

**Pull request merged**